File: implementation/scipamato/core/sync/src/adhocTest/java/ch/difty/scipamato/core/sync/launcher/RefDataSyncJobLauncherAdHocTest.java
package ch.difty.scipamato.core.sync.launcher;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest
class RefDataSyncJobLauncherAdHocTest {

    @Autowired
    private RefDataSyncJobLauncher launcher;

    @Test
    void run() {
        SyncJobResult result = launcher.launch();
        result
            .getMessages()
            .forEach(System.out::println);
        assertThat(result.isSuccessful()).isTrue();
    }
}

File: core/core-sync/src/adhoc-test/kotlin/ch/difty/scipamato/core/sync/launcher/RefDataSyncJobLauncherAdHocTest.kt
package ch.difty.scipamato.core.sync.launcher

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
internal open class RefDataSyncJobLauncherAdHocTest {

    @Autowired
    private lateinit var launcher: RefDataSyncJobLauncher

    @Test
    fun run() {
        val result = launcher.launch()
        result.messages.forEach { println(it) }
        assertThat(result.isSuccessful).isTrue()
    }
}
